tests/end_to_end/CairoPrecompiles/test_batch_cairo_precompile.py

```diff
@@ -0,0 +1,147 @@
+from typing import Any, List, Tuple
+
+import pytest
+import pytest_asyncio
+from eth_abi import encode
+from eth_utils import keccak
+from hypothesis import given, settings
+from hypothesis import strategies as st
+
+from kakarot_scripts.utils.kakarot import deploy, eth_send_transaction
+from kakarot_scripts.utils.starknet import get_contract, wait_for_transaction
+from tests.utils.errors import cairo_error
+
+EVM_MULTICALLCAIRO_SELECTOR = keccak(
+    text="call_contract(uint256,uint256,uint256,uint256[])"
+)[:4]
+
+
+@pytest_asyncio.fixture(scope="module")
+async def cairo_counter(max_fee, deployer):
+    cairo_counter = get_contract("Counter", provider=deployer)
+
+    yield cairo_counter
+
+    tx = await cairo_counter.functions["set_counter"].invoke_v1(0, max_fee=max_fee)
```

Comment:
it's better to use our wrapper `call` and `invoke` as starknet py is not so stable, and so it's easier when we need to update/patch

Reply:
call and invoke are good for contracts available in artifacts, not so much when passing an address because the calldata  is not parsed.

Reply:
if `get_contract` gives you the right contract without providing `address=`, then it means that `call` and `invoke` will target the right contract by default

Reply:
ah yes ok i got it now!